nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCPRecord.java

```diff
@@ -420,12 +385,12 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                     getLogger().debug("Removing flow file, no records were written");
                     session.remove(flowFile);
                 } else {
-                    final String sender = getRemoteAddress(socketRecordReader);
+                    final String sender = recordReader.getSender().toString();
 
                     final Map<String, String> attributes = new HashMap<>(writeResult.getAttributes());
                     attributes.put(CoreAttributes.MIME_TYPE.key(), mimeType);
                     attributes.put("tcp.sender", sender);
-                    attributes.put("tcp.port", String.valueOf(port));
+                    attributes.put("tcp.port", String.valueOf(port)); // Should this be the remote port..?
```

Comment:
The docs for tcp.port say

@WritesAttribute(attribute="tcp.port", description="The port that the processor accepted the connection on."),

But possibly we should add another attribute like tcp.sender.port

Reply:
I agree that a new attribute for the sender port could be useful.

Reply:
That's fine, I didn't check the WritesAttribute. Right now the sender will look something like "/127.0.0.1:63473". Knowing this, is it necessary to add the sender port separately? I can compare the sender attribute to before these changes and make sure that it's still formatted the same for backwards compatibility.

Reply:
Seeing that we already include the `tcp.sender` property, adding a new attribute for the port seems unnecessary, thanks for clarifying.